almdrlib/environment.py

```diff
@@ -54,27 +55,36 @@ class AlEnvAwsConfigurationException(AlEnvException):
 class AlEnvConfigurationTableUnavailableException(AlEnvException):
     pass
 
+class AlmdrlibSourceNotEnabledError(AlEnvException):
+    pass
+
 
 class AlEnv:
-    def __init__(self, application_name):
+    def __init__(self, application_name, client=None, source="dynamodb"):
```

Comment:
Why restrict an instance of AlEnv to ssm or ddb, why not support both?

Reply:
This doesn't really change how it is used it in config.py though - we need to create an ssm-only AlEnv because creation will fail for one with (dynamodb, ssm) if the lambda using it doesn't have permission to read the settings table (and we wouldn't give it those permissions if the keys are in ssm)

Reply:
This also means it would fail if the service does not have permissions to read the info from DDB. We should just handle both not found and lack of permissions in both cases.

Reply:
I assume if someone is using AlEnv directly then they will only create an instance with both sources enabled if they know their service has permissions for both. Otherwise they should just create one with either ssm or dynamodb enabled.

What do you mean by "handle not found and lack of permissions in both cases"? Are you thinking we should catch all errors in __init__, and if there was a setup error in dynamodb, just fail when some client calls get?

Reply:
Ahh no, I mean try both if needed and if both fails and no credentials are set up at the end of init then we throw the error, vs throwing a error for no permissions to read from ssm/ddb.